jsonschema-core.xml

```diff
@@ -1066,6 +1066,94 @@
             </t>
         </section>
 
+        <section title="Collecting Annotations">
+            <t>
+                <cref>
+                    The exact structure and format of the information collected is TBD,
+                    but will be defined before the next draft.  Some details of this
+                    section may change as a result, but the overall process is expected
+                    to remain the same.  See GitHub issue #396 to track progress.
+                </cref>
+            </t>
+            <t>
+                Annotations are collected by keywords that explicitly define
+                annotation-collecting behavior.  Note that boolean schemas cannot
+                produce annotations as they do not make use of keywords.
+            </t>
+            <t>
+                A collected annotation MUST include the following information:
+                <list>
+                    <t>
+                        The name of the annotation, which MUST be identical to the keyword
+                        that defines that annotation.
+                    </t>
+                    <t>
+                        The instance location to which it is attached, as a JSON Pointer
+                    </t>
+                    <t>
+                        The schema location of the attaching keyword, as a list of URIs
+                        constructed as described below
+                    </t>
+                    <t>
+                        The attached value(s)
+                    </t>
+                </list>
+            </t>
+            <t>
+                If multiple schema locations attach values to the same instance location,
+                and the annotation defines a process for combining such values, then the
+                combined value MUST also be associated with the instance location.
+            </t>
+            <t>
+                Applications MAY make decisions on which of multiple annotation values
+                to use based on the schema location that contributed the value.
+                This is intended to allow flexible usage.  Collecting the schema location
+                facilites such usage.
+            </t>
+            <t>
+                For example, one application may consider a "description" annotation that is
+                in the same schema object as a "$ref" to override any "description" in the
+                reference's target schema.  A different application may prefer to concatenate
+                all "description" annotations based on whatever ordering it defines.
+            </t>
+            <section title="Annotations and Assertions">
+                <t>
+                    If any keyword in a schema object produces a false assertion
+                    result, then all annotations from all keywords in that schema
+                    object, and any of its subschemas or referenced schemas, MUST
+                    be discarded.
```

Comment:
This could be shortened for clarity, I think.
"Annotations are only generated from instances that validate against the schema."

And care is needed so we don't confuse people implementing "not". The note about "not" is only (and should only be) non-normative.

Reply:
@awwright I'm going to remove most of the "Annotations and Applicators" section to get rid of the details around `not` and similar cases, and try to clarify why I'm using the "must be discarded" language in the "Annotations and Assertions" section.